shared/js/otSdkDriver.js

```diff
@@ -1001,20 +1001,28 @@ loop.OTSdkDriver = (function() {
         // Nothing to do for the success case.
         return;
       }
-      if (!(error.message && error.message === "DENIED")) {
-        // We free up the publisher here in case the store wants to try
-        // grabbing the media again.
-        if (this.publisher) {
-          this.publisher.off("accessAllowed accessDenied accessDialogOpened streamCreated");
-          this.publisher.destroy();
-          delete this.publisher;
-          delete this._mockPublisherEl;
-        }
-        this.dispatcher.dispatch(new sharedActions.ConnectionFailure({
-          reason: FAILURE_DETAILS.UNABLE_TO_PUBLISH_MEDIA
-        }));
-        this._notifyMetricsEvent("sdk.exception." + error.code + "." + error.message);
+
+      if (error.message && error.message === "DENIED") {
+        // In the DENIED case, this will be handled by _onPublishDenied.
+        return;
       }
+
+      // We free up the publisher here in case the store wants to try
+      // grabbing the media again.
```

Comment:
This comment seems more relevant a few lines lower. Or change the intro as "Only free up the publisher if we have one in case ..."